### ISSUE TITLE
Implement concept for Modal windows

### ISSUE THREAD

**Issue by barmac** [camunda/camunda-modeler#1681]
This is a status issue for what I am working on at the moment.

__What should we do?__

* [x] Location: It needs to appear upper half of the screen, 110px from the top and 60px from the bottom per default in a 1440x1024 screen. As the size of the viewport changes, the modal window will be centered in the viewport accordingly and the top position of the modal will be relocated.
* [x] Resizing: If the modal window needs to grow for adding more elements, then the modal window height will be increased from the bottom. Top position of the modal window will not change by adding more elements in the modal.
* [x] Scrolling: When the modal window hit the maximum distance to the bottom of the viewport then the content will be scrollable. The modal window itself should be visible all the time (header, content and buttons area).
* [x] Size: Width of the window will be static and be 636px [this is not a final value yet]. Height of the window will be dynamic.
* [x] Closing: User can close the window by clicking Close icon (X), Close / Cancel button and  pressing ESC key. Dismissible modals (click on background closes modal) should not be allowed.
* [x] Submitting: Primary action should be triggered by Enter key if the focus in on the button.
* [x] Buttons: If only one option exists, this option should be primary button ("Close") and can be triggered by ESC / Enter buttons.
* [x] Focus: 
  * [x] Trap: Keyboard focus needs to stay in the modal window, focus should not reach elements behind the modal window.
  * [x] Background: While modal window is visible, darken the background with a transparent overlay to help user to concentrate on modal.
* [x] Implement new look
  * [x] layout

Some of the items are already implemented.

__Why should we do it?__

We want modals to look and feel good.

---

Child of https://github.com/camunda/camunda-modeler/issues/1609

**Comment by nikku** [comment]
Could you link the design screens for reference?

**Comment by barmac** [comment]
> Could you link the design screens for reference?

Links are available here: https://confluence.camunda.com/pages/viewpage.action?spaceKey=PD&title=Modal+Windows+in+Modeler

**Comment by barmac** [comment]
Closed via https://github.com/camunda/camunda-modeler/pull/1686
